test/functional/Java14andUp/src/org/openj9/test/java/lang/Test_Thread.java

```diff
@@ -0,0 +1,203 @@
+ */
+
+@Test(groups = { "level.sanity" })
+public class Test_Thread {
+
+    /* interrupt thread before it has started */
+    @Test
+    public void test_interruptBeforeRun() {
+        String name = "interruptBeforeRun: ";
+        Thread t = new Thread();
+
+        /* Verify thread is dead and has not been interrupted */
+        AssertJUnit.assertFalse(name + "thread should not be alive", t.isAlive());
+        AssertJUnit.assertFalse(name + "interrupt flag should not be set", t.isInterrupted());
+
+        t.interrupt();
+
+        /* Verify that the thread was successfully interrupted. */
+        AssertJUnit.assertTrue(name + "thread that has not started should have interrupt flag set", t.isInterrupted());
+    }
+
+    /* interrupt thread after it has ended */
+    @Test
+    public void test_interruptAfterRun() throws Throwable {
+        String name = "interruptAfterRun: ";
+        Thread t = new Thread();
+
+        /* run thread */
+        t.start();
+        t.join();
+
+        /* Verify thread is dead and has not been interrupted */
+        AssertJUnit.assertFalse(name + "thread should not be alive", t.isAlive());
+        AssertJUnit.assertFalse(name + "interrupt flag should not be set", t.isInterrupted());
+
+       t.interrupt();
+
+        /* Verify that the thread was successfully interrupted. */
+        AssertJUnit.assertTrue(name + "thread that has ended was was interrupted", t.isInterrupted());    
+    }
+
+    /* if thread is interrupted before start, verify that flag is still set after run. */
+    @Test
+    public void test_interruptAtStartSetAfterRun() throws Throwable {
+        String name = "interruptAtStartSetAfterRun: ";
+        Thread t = new Thread();
+
+        /* Verify thread is dead and has not been interrupted */
+        AssertJUnit.assertFalse(name + "thread should not be alive", t.isAlive());
+        AssertJUnit.assertFalse(name + "interrupt flag should not be set", t.isInterrupted());
+
+        t.interrupt();
+
+        /* run thread */
+        t.run();
+        t.join();
+
+        /* Verify thread is dead and has not been interrupted */
+        AssertJUnit.assertFalse(name + "thread should be dead", t.isAlive());
+        AssertJUnit.assertTrue(name + "interrupt flag should be set", t.isInterrupted()); 
+    }
+
+    /* Interrupt thread during run, verify that interrupt flag is not set when thread stops running. 
+     * isInterrupted becomes false after InterruptedException is thrown. */
+    @Test
+    public void test_interruptDuringRun() throws Throwable {
+        String name = "interruptDuringRun: ";
+        Thread t = new Thread() {
+            public void run() {
+                synchronized(this) {
+                    try {
+                        wait();
+                        AssertJUnit.fail(); /* not interrupted */
+                    } catch (InterruptedException e) {
+                        /* expected */
```

Comment:
Pls set a volatile boolean field here, and test it after join(). Similarly in the next test.